python/ray/serve/_private/replica.py

```diff
@@ -665,22 +669,34 @@ async def perform_graceful_shutdown(self):
 
         self._metrics_manager.shutdown()
 
-    @ray.method(concurrency_group=REPLICA_CONTROL_PLANE_CONCURRENCY_GROUP)
     async def check_health(self):
-        await self._user_callable_wrapper.call_user_health_check()
+        # If the user provided a health check, call it via `user_callable_wrapper` on
+        # the user code thread. This means if user code blocks the event loop the health
+        # check may time out.
+        #
+        # To avoid this issue for basic cases without a user-defined health check, skip
+        # interacting with the `user_callable_wrapper` entirely.
+        if self._user_callable_wrapper.has_user_health_check:
```

Comment:
Can we push this conditional into wrapper itself?

Reply:
good idea I think I'll need to do that for unit testing anyways